src/validator/SocialRecoveryValidator.sol

```diff
@@ -255,15 +257,28 @@ contract SocialRecoveryValidator is IKernelValidator {
                 if (owner != recovered) {
                     return SIG_VALIDATION_FAILED;
                 }
-            } else if(mode == bytes1(0x02)) {
-                address kernelAddress = address(bytes20(_userOp.callData[233:253]));
-                require(isGuardian[kernelAddress][_userOp.sender],"Recovery Plugin: Not a guardian");
+            } else if (mode == bytes1(0x02)) {
+                address kernelAddress = address(
+                    bytes20(_userOp.callData[233:253])
+                );
+                require(
+                    isGuardian[kernelAddress][_userOp.sender],
```

Comment:
Not sure if this is ok, kernelAddress==userOp.sender right?

Reply:
Won't be in this mode, since kernelAddress is the address of the account which is to be recovered and userOp.sender is any guardian so kernelAddress!=userOp.sender

Reply:
so you mean guardian is another kernel?